sporkfish/searcher/searcher_factory.py

```diff
@@ -12,30 +11,7 @@ class SearcherFactory:
     """
 
     @staticmethod
-    def _build_evaluator(
-        evaluator_type: EvaluateMode = EvaluateMode.PESTO,
-    ) -> Evaluator:
-        """
-        Build and return an instance of Evaluator based on the specified evaluation mode.
-
-        :param evaluator_type: The type of evaluation to use.
-        :type evaluator_type: EvaluateMode
-        :return: An instance of Evaluator.
-        :rtype: Evaluator
-        :raises TypeError: If the specified evaluator type is not supported.
-        """
-        # this method needs changing when refactoring Evaluator design, see:
-        # https://github.com/KYLChiu/sporkfish/issues/88
-        if evaluator_type is EvaluateMode.PESTO:
-            return Evaluator()
-        else:
-            raise TypeError(
-                f"SearcherFactory does not support the creation of Evaluator type: \
-                {type(evaluator_type).__name__}."
-            )
-
-    @staticmethod
-    def create(searcher_cfg: SearcherConfig) -> Searcher:
+    def create(searcher_cfg: SearcherConfig, evaluator: Evaluator) -> Searcher:
```

Comment:
I know this is easier but I'd prefer if we keep evaluator out of the factory parameter. The reason is that other searchers (e.g. MCTS) don't need a evaluator, so I'd rather keep a consistent interface. With the suggestion to put evaluator config in searcher config before I think it should work, we can call the evaluator factory where its needed (i.e. in the creation of those two searchers). How does this sound?

Reply:
EDIT: I guess we can pass None if it doesn't need it. Let's just keep it like this for now.